source/server.js

```diff
@@ -0,0 +1,192 @@
+/* Import express */
+const express = require('express');
+
+/* Create express server instance */
+const app = express();
+
+let scores;
+
+app.use(express.static('./source/frontend'));
+app.use(express.json());
+
+const fs = require('fs').promises;
+
+let turnCount = 0;
+
+const board = [
+  [0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0],
+];
+
+/* Reads from file and adds contents to global variable 'scores' */
+async function readScores() {
+  const rawData = await fs.readFile('./source/data/game.json', 'utf-8');
+  const parsedElement = JSON.parse(rawData);
+  scores = parsedElement;
+}
+
+/* Selects the element of the file that matches the given player
+   and updates its score */
+async function updateScore(player) {
+  try {
+    await readScores();
+
+    const playerObj = scores.filter((item) => item.player === player)[0];
+
+    playerObj.score += 1;
+
+    await fs.writeFile('./source/data/game.json', JSON.stringify(scores, null, '  '), 'utf-8');
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+function isPositionTaken(_board, xPos, yPos) {
+  if (xPos === undefined || yPos === undefined) {
+    throw new Error('Argument undefined');
+  }
+
+  return _board[yPos][xPos] !== 0;
+}
+
+function dropToBottom(_board, xPos, freePos) {
```

Comment:
This function accepts an argument of `freePos`, but that argument isn't used except to hold a value on line 61 (and even then this is not necessary as you could `return j`). I suggest it should be removed
```suggestion
function dropToBottom(_board, xPos) {
```